File: projects/m1/011-note-to-frequency/solutions/GiorgioCignitti/noteToFrequency.py
note=input("insert a note (i.e. C4): ")

def calcFrequency(note):

  letter=note[0]
  number=int(note[1])

  if(letter=="C"):
    frequency=261.33/ 2**(4 - number)
  elif(letter=="D"):
     frequency=293.66/ 2**(4 - number)
  elif(letter=="E"):
     frequency+=329.63/ 2**(4 - number)
  elif(letter=="F"):
     frequency+=349.23/ 2**(4 - number)
  elif(letter=="G"):
     frequency+=392.00/ 2**(4 - number)
  elif(letter=="A"):
     frequency+=440.00/ 2**(4 - number)
  elif(letter=="B"):
     frequency+=493.88/ 2**(4 - number)
  
  return f"Frequency of your note is {frequency} "

print(calcFrequency(note))

File: projects/m1/011-note-to-frequency/solutions/stefandifede/note-to-frequency.js
const noteInput = prompt("Insert a note")

//creo un array delle note in oggetti
class notes {
    constructor(noteName, frequency){
        this.noteName = noteName;
        this.frequency = frequency
    }
}

const c = new notes ("c", 261.63) 
const d = new notes ("d", 293.66)
const e = new notes ("e", 329,63)
const f = new notes ("f", 349.23)
const g = new notes ("g", 392.00)
const a = new notes ("a", 440.00)
const b = new notes ("b", 493.88)

const notesArr = [c, d, e, f, g, a, b]

//dividere l'input in lettera e numero

let arr = []
arr = noteInput.split('')


//ottenere la frequenza richiesta partendo dalla frequenza 4 della lettera inserita

function findFrequency () {
    for (i = 0; i <= notesArr.length - 1; i++) {
        if (arr[0] == notesArr[i].noteName) {
            alert((notesArr[i].frequency/(Math.pow(2, 4-arr[1]))).toFixed(2) + " hz")
        }
    }
}

findFrequency()